pychromecast/__init__.py

```diff
@@ -194,6 +197,13 @@ def __init__(self, host, port=None, device=None, **kwargs):
         if blocking:
             self.socket_client.start()
 
+        # Set to True by default if cast_type == CAST_TYPE_GROUP
+        self.dynamic_host = True if self.cast_type == CAST_TYPE_GROUP \
+            else False
+        self.dynamic_host = kwargs.pop('dynamic_host', self.dynamic_host)
+        if self.dynamic_host:
+            self.start_dynamic_host_tracking()
```

Comment:
We don't want to create a new thread for each cast group. I think what we'd have to do is start the thread for the first group, and add each listening object id to a list. The objects can remove themselves from the list in `__del__`, and when the list is empty the thread can be stopped.